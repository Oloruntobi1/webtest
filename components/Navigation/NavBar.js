import NavLink from './NavLink';

const NAV_LINKS = [
  {
    id: 1,
    title: 'Job Openings',
    anchor: '/jobs',
  },
  {
    id: 2,
    title: 'About Us',
    anchor: '/#about',
  },
  {
    id: 3,
    title: 'Finding Talent',
    anchor: '/finding-talent',
  },
  {
    id: 4,
    title: 'Contact',
    anchor: '/#contact',
  },
];

export default function NavBar() {
  return (
    <div
      id="navBar"
      className="collapse navbar-collapse navbar-nav-wrap-collapse"
    >
      <div className="navbar-body header-abs-top-inner py-3">
        <ul className="js-scroll-nav navbar-nav header-navbar-nav">
          {NAV_LINKS.map((link) => (
            <NavLink link={link} key={link.id} />
          ))}
        </ul>
      </div>
    </div>
  );
}
