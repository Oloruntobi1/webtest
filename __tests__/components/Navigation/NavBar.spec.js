import React from 'react';
import renderer from 'react-test-renderer';

import * as nextRouter from 'next/router';

import NavBar from '../../../components/Navigation/NavBar';

nextRouter.useRouter = jest.fn();
nextRouter.useRouter.mockImplementation(() => ({ route: '/' }));

it('renders correctly', () => {
  const tree = renderer
    .create(<NavBar />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
