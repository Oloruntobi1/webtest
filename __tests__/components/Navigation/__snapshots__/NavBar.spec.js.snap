// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="collapse navbar-collapse navbar-nav-wrap-collapse"
  id="navBar"
>
  <div
    className="navbar-body header-abs-top-inner py-3"
  >
    <ul
      className="js-scroll-nav navbar-nav header-navbar-nav"
    >
      <li
        className="navbar-nav-item mx-lg-4 navItem"
      >
        <a
          className="nav-link font-weight-normal text-center px-0 py-2 navLink"
          href="/jobs"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Job Openings
        </a>
      </li>
      <li
        className="navbar-nav-item mx-lg-4 navItem"
      >
        <a
          className="nav-link font-weight-normal text-center px-0 py-2 navLink"
          href="/#about"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          About Us
        </a>
      </li>
      <li
        className="navbar-nav-item mx-lg-4 navItem"
      >
        <a
          className="nav-link font-weight-normal text-center px-0 py-2 navLink"
          href="/finding-talent"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Finding Talent
        </a>
      </li>
      <li
        className="navbar-nav-item mx-lg-4 navItem"
      >
        <a
          className="nav-link font-weight-normal text-center px-0 py-2 navLink"
          href="/#contact"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Contact
        </a>
      </li>
    </ul>
  </div>
</div>
`;
